dask/array/tests/test_reductions.py

```diff
@@ -422,3 +422,45 @@ def test_array_cumreduction_out(func):
     x = da.ones((10, 10), chunks=(4, 4))
     func(x, axis=0, out=x)
     assert_eq(x, func(np.ones((10, 10)), axis=0))
+
+
+def test_topk_argtopk():
+    # Test data
+    k = 5
+    a = da.random.random(1000, chunks=250)
+    b = da.random.random((10, 20, 30), chunks=(4, 8, 8))
+    c = da.from_array(np.array([(1, 'Hello'), (2, 'World')], dtype=[('foo', int), ('bar', '<U5')]),
+                      chunks=1)
+
+    # Support for deprecated API for topk
+    np.testing.assert_array_equal(da.topk(a, 5), da.topk(5, a))
+
+    # As Array methods
+    a.topk(5)
+    np.testing.assert_array_equal(da.topk(a, 5), da.topk(5, a))
```

Comment:
In dask we make use of `assert_eq` to test equality. This checks not only values, but also the metadata on the dask arrays, which helps ensure correctness. You should be able to replace all your calls to `assert_array_equal` with this.